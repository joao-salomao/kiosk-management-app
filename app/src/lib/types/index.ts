export type Kiosk = {
  id: string;
  serialKey: string;
  description: string;
  isKioskClosed: boolean;
  storeOpensAt: string;
  storeClosesAt: string;
};

export enum StatusFilterEnum {
  open = "open",
  closed = "closed",
  all = "all",
}
