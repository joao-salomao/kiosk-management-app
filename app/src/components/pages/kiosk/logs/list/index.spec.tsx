import { Log } from 'lib/types';
import renderer from 'react-test-renderer';
import { RecoilRoot } from 'recoil';
import { List } from ".";

it('renders logs list correctly', () => {
  const logs: Log[] = [
    {
      id: "1",
      action: "Create kiosk",
      description: JSON.stringify({ from: {}, to: {} }),
      kiosk: {
        id: "1",
        description: "Mac",
        isKioskClosed: false,
        serialKey: "1234",
        storeOpensAt: "06:00",
        storeClosesAt: "23:00",
      },
      user: {
        id: "1",
        name: "Some name",
        email: "email@email.com",
      }
    },
    {
      id: "2",
      action: "Update kiosk",
      description: JSON.stringify({
        from: {
          storeClosesAt: "23:00"
        }, to: {
          storeClosesAt: "23:30"
        }
      }),
      kiosk: {
        id: "1",
        description: "Mac",
        isKioskClosed: false,
        serialKey: "1234",
        storeOpensAt: "06:00",
        storeClosesAt: "23:30",
      },
      user: {
        id: "1",
        name: "Some name",
        email: "email@email.com",
      }
    },
  ]

  const tree = renderer
    .create(
      <RecoilRoot>
        <List
          logs={logs}
          isLoading={false}
        />
      </RecoilRoot>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});


it('renders loader correctly', () => {
  const tree = renderer
    .create(
      <RecoilRoot>
        <List
          logs={[]}
          isLoading={true}
        />
      </RecoilRoot>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});