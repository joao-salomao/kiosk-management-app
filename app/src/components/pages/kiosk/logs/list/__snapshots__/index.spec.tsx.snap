// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders logs list correctly 1`] = `
<div
  className="text-gray-500 shadow-lg sm:rounded-lg p-4 min-h-[150px]"
>
  <h5
    className="mb-2 text-2xl font-bold tracking-tight text-gray-700"
  >
    Logs
  </h5>
  <div
    className="mb-2 max-w-[200px]"
  >
    <div
      className="mr-4"
    >
      <label
        htmlFor="searchFilter"
        id="search-label"
      >
        Search
      </label>
      <br />
      <div
        className="relative w-full"
      >
        <div
          className="flex absolute inset-y-0 left-0 items-center pl-3 pointer-events-none"
        >
          <svg
            className="w-5 h-5 text-gray-500 dark:text-gray-400"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z"
              fillRule="evenodd"
            />
          </svg>
        </div>
        <input
          aria-labelledby="search-label"
          className="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full pl-10 p-2"
          name="searchFilter"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="block p-4 bg-white rounded-lg border border-gray-200 shadow-sm hover:bg-gray-100 mb-3"
  >
    <h5
      className="text-md font-bold tracking-tight"
    >
      Log #
      1
    </h5>
    <div>
      <span
        className="font-bold"
      >
        Kiosk:
      </span>
       #
      1
    </div>
    <div>
      <span
        className="font-bold"
      >
        User:
      </span>
       
      Some name
    </div>
    <div>
      <span
        className="font-bold"
      >
        Action:
      </span>
       
      Create kiosk
    </div>
  </div>
  <div
    className="block p-4 bg-white rounded-lg border border-gray-200 shadow-sm hover:bg-gray-100 mb-3"
  >
    <h5
      className="text-md font-bold tracking-tight"
    >
      Log #
      2
    </h5>
    <div>
      <span
        className="font-bold"
      >
        Kiosk:
      </span>
       #
      1
    </div>
    <div>
      <span
        className="font-bold"
      >
        User:
      </span>
       
      Some name
    </div>
    <div>
      <span
        className="font-bold"
      >
        Action:
      </span>
       
      Update kiosk
    </div>
    <div>
      <p
        className="font-bold"
      >
        Changes:
      </p>
      <div>
        <span
          className="underline"
        >
          #storeClosesAt
        </span>
         from "
        23:00
        " to "
        23:30
        "
      </div>
    </div>
  </div>
</div>
`;

exports[`renders the empty list message when there is no logs to show 1`] = `
<div
  className="text-gray-500 shadow-lg sm:rounded-lg p-4 min-h-[150px]"
>
  <h5
    className="mb-2 text-2xl font-bold tracking-tight text-gray-700"
  >
    Logs
  </h5>
  <div
    className="mb-2 max-w-[200px]"
  >
    <div
      className="mr-4"
    >
      <label
        htmlFor="searchFilter"
        id="search-label"
      >
        Search
      </label>
      <br />
      <div
        className="relative w-full"
      >
        <div
          className="flex absolute inset-y-0 left-0 items-center pl-3 pointer-events-none"
        >
          <svg
            className="w-5 h-5 text-gray-500 dark:text-gray-400"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z"
              fillRule="evenodd"
            />
          </svg>
        </div>
        <input
          aria-labelledby="search-label"
          className="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full pl-10 p-2"
          name="searchFilter"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="px-6 py-4 text-center"
  >
    No results to show, try to change the filters or create a new kiosk.
  </div>
</div>
`;

exports[`renders the loader correctly 1`] = `
<div
  className="text-gray-500 shadow-lg sm:rounded-lg p-4 min-h-[150px]"
>
  <div
    className="h-20 w-20 mx-auto"
  >
    <p
      className="mb-2"
    >
      Carregando
    </p>
    <svg
      className="inline text-gray-200 animate-spin fill-blue-600"
      fill="none"
      role="status"
      viewBox="0 0 100 101"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M100 50.5908C100 78.2051 77.6142 100.591 50 100.591C22.3858 100.591 0 78.2051 0 50.5908C0 22.9766 22.3858 0.59082 50 0.59082C77.6142 0.59082 100 22.9766 100 50.5908ZM9.08144 50.5908C9.08144 73.1895 27.4013 91.5094 50 91.5094C72.5987 91.5094 90.9186 73.1895 90.9186 50.5908C90.9186 27.9921 72.5987 9.67226 50 9.67226C27.4013 9.67226 9.08144 27.9921 9.08144 50.5908Z"
        fill="currentColor"
      />
      <path
        d="M93.9676 39.0409C96.393 38.4038 97.8624 35.9116 97.0079 33.5539C95.2932 28.8227 92.871 24.3692 89.8167 20.348C85.8452 15.1192 80.8826 10.7238 75.2124 7.41289C69.5422 4.10194 63.2754 1.94025 56.7698 1.05124C51.7666 0.367541 46.6976 0.446843 41.7345 1.27873C39.2613 1.69328 37.813 4.19778 38.4501 6.62326C39.0873 9.04874 41.5694 10.4717 44.0505 10.1071C47.8511 9.54855 51.7191 9.52689 55.5402 10.0491C60.8642 10.7766 65.9928 12.5457 70.6331 15.2552C75.2735 17.9648 79.3347 21.5619 82.5849 25.841C84.9175 28.9121 86.7997 32.2913 88.1811 35.8758C89.083 38.2158 91.5421 39.6781 93.9676 39.0409Z"
        fill="currentFill"
      />
    </svg>
  </div>
</div>
`;
